addons/udes_stock_refactoring/tests/test_splitting.py

```diff
@@ -0,0 +1,649 @@
+        Assert that the supplied picking recordset has only 1 record.
+        Assert that the state, origin, partner, group name, location,
+        destination location and date done match the expected values 
+        if supplied.
+        """
+        self.assertEqual(len(picking), 1)
+
+        if state:
+            self.assertEqual(picking.state, state)
+        if origin:
+            self.assertEqual(picking.origin, origin)
+        if partner:
+            self.assertEqual(picking.partner_id, partner)
+        if group_name:
+            self.assertEqual(picking.group_id.name, group_name)
+        if location_id:
+            self.assertEqual(picking.location_id, location_id)
+        if location_dest_id:
+            self.assertEqual(picking.location_dest_id, location_dest_id)
+        if date_done:
+            self.assertEqual(picking.date_done, date_done)
+
+
+class TestAssignSplitting(TestSplittingBase):
+    def setUp(self):
+        """
+        Set post assign action for test pick type
+        """
+        # group by package post assign
+        self.picking_type_pick.write(
+            {
+                "u_post_assign_action": "group_by_move_line_key",
+                "u_move_line_key_format": "{package_id.name}",
+            }
+        )
+
+        super(TestAssignSplitting, self).setUp()
+
+    def test_reserve_1_pallet_diff_prods_splits_1_pick_per_pallet(self):
+        """
+        With 1 pallet each for 2 different products.
+        Reserve pick (action_assign).
+        Pick should be split into 2 picks 1 for each pallet, 
+        reusing the original pick.
+        """
+        apple_pallet = self.apple_pallet
+        banana_pallet = self.banana_pallet
+
+        products_info = [{"product": self.apple, "qty": 10}, {"product": self.banana, "qty": 10}]
+        moves = self.create_move(self.picking, products_info)
+
+        apple_move = moves.filtered(lambda m: m.product_id == self.apple)
+        banana_move = moves - apple_move
+
+        self.picking.action_assign()
+
+        apple_pick = apple_move.picking_id
+        banana_pick = banana_move.picking_id
+
+        self._assert_picking_fields(apple_pick, state="assigned", group_name=apple_pallet.name)
+        self._assert_picking_fields(banana_pick, state="assigned", group_name=banana_pallet.name)
+
+        # Check we haven't mangled the moves or move_lines
+        apple_move = apple_pick.move_lines
+        self._assert_move_fields(apple_move, self.apple, 10)
+
+        apple_move_line = apple_pick.move_line_ids
+        self._assert_move_line_fields(apple_move_line, self.apple, 10, apple_pallet)
+
+        banana_move = banana_pick.move_lines
+        self._assert_move_fields(banana_move, self.banana, 10)
+
+        banana_move_line = banana_pick.move_line_ids
+        self._assert_move_line_fields(banana_move_line, self.banana, 10, banana_pallet)
+
+        # Check that the original pick has been reused
+        self.assertTrue(self.picking.exists())
```

Comment:
You could also check that the original picking is either apple_pick or banana_pick.

Reply:
Have now added a check to make sure that `self.picking` is either `apple_pick` or `banana_pick`.